src/Utilities/List.f90

```diff
@@ -285,17 +254,15 @@ subroutine InsertAfter(this, objptr, indx)
         followingNode%prevNode => newNode
         this%nodeCount = this%nodeCount + 1
       else
-        write (line, '(a)') 'Programming error in ListType%insert_after'
-        call sim_message(line)
-        call stop_with_error(1)
+        print *, 'Programming error in ListType%insert_after'
+        error stop 1
```

Comment:
We might need to check further on the ramifications of this change.  I'm thinking stop_with_error() and its use of call exit() was used intentionally, perhaps for API behavior.

Reply:
Since gnu, intel, and flang [all support](https://flang.llvm.org/docs/Extensions.html) an `exit()` intrinsic subroutine maybe it makes sense to standardize on that directly, and `stop_with_error()` would be unnecessary

One drawback of `error stop` is it only supports a variable status code in f2018, while `exit()` already does

Reply:
Would it make sense to keep `stop_with_error()` in place and use it for all programming errors?  It could remain the same or we could pass it a message.  If we ever needed to change the behavior, we would only need to change it one place?  Or we could get rid of it and just call exit().

Reply:
> Would it make sense to keep stop_with_error() in place and use it for all programming errors?  It could remain the same or we could pass it a message. If we ever needed to change the behavior, we would only need to change it one place?

That seems good to me. Maybe it could live in a separate module so generic utilities don't depend on simulation-related things?

Reply:
Sounds good to me!